src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala

```diff
@@ -738,6 +738,7 @@ sealed trait CloudStorage {
           .setSnowflakeFileTransferMetadata(fileTransferMetadata.get)
           .setUploadStream(inStream)
           .setRequireCompress(compress)
+          .setDestFileName(fileName)
```

Comment:
For uploading multiple partitions, we only use ` PUT` command to get uploading one metadata, So we need to set `destFileName` for each file uploading. 
PS: `SnowflakeFileTransferAgent.uploadWithoutConnection()` has been enhanced to use the DestFileName.